test/Basics/VerifySkipNestedDeferred.baseline

```diff
@@ -12,6 +12,8 @@ SkipNestedDeferred:	Pushing a reference to 'x'
 Pass: Nested unnamed function expression
 Skipping nested deferred function 11. a1: 1438...1546
 SkipNestedDeferred:	Pushing a reference to 'x'
+Skipping nested deferred function 12. a2: 1469...1517
+SkipNestedDeferred:	Pushing a reference to 'x'

```

Comment:
@boingoing can you take a look at this? I'm not sure what exactly this test is intending to do, and I wasn't expecting to make this kind of change, but it looks plausible to me...

Reply:
This test is just verifying that when we have the parser state cache we can use it to skip scanning nested functions. What's going on here is we are skipping scan for a function which we weren't before. I guess this is because of the new flag you added for async functions in defer parse which is making us go down a different path in the parser. I didn't expect that but the test behavior here is not wrong so it should be fine as long as other tests are also passing.